### PR TITLE
Sw 121

### PR DESCRIPTION
**Pull request by Thekid2002**

### CODE REVIEW

File: Models/Suppliers/Supplier.cs [hunk 1]
```diff
@@ -1,34 +1,55 @@
...
+
+    public async Task<List<int>> GetAssociatedItemsAsIntList(SharedContext context)
+    {
+        return await context.Items
+            .Where(item => context.SupplierItemRelations.Any(sir => sir.SupplierId == this.Id && sir.ItemId == item.Id))
+            .Select(item => item.Id)
+            .ToListAsync();
     }
+
+    public async Task ClearAssociatedItems(SharedContext context)
```
Comment (ChristofferGamel) [inline Models/Suppliers/Supplier.cs:72]: comments who?

File: Models/Suppliers/Supplier.cs [hunk 2]
```diff
@@ -1,34 +1,55 @@
...
+    }
+
+    public async Task<List<Item>> GetAssociatedItems(SharedContext context)
+    {
+        return await context.Items
+            .Where(item => context.SupplierItemRelations.Any(sir => sir.SupplierId == this.Id && sir.ItemId == item.Id))
+            .ToListAsync();
+    }
+
+    public async Task<List<int>> GetAssociatedItemsAsIntList(SharedContext context)
```
Comment (ChristofferGamel) [inline Models/Suppliers/Supplier.cs:60]: comments what?

File: Controllers/Shared/ItemController.cs [hunk 3]
```diff
@@ -69,5 +71,21 @@ public async Task<ActionResult<ItemDto>> PutItem([FromBody] ItemDto item)
                 return BadRequest(e.Message);
             }
         }
+
+        [HttpDelete("{id}")]
+        [Authorize(Policy = "require-admin-role")]
```
Comment (ChristofferGamel) [inline Controllers/Shared/ItemController.cs]: Items er ikke admin required ifølge class diagram

image

File: Models/Suppliers/Supplier.cs [hunk 4]
```diff
@@ -1,34 +1,55 @@
...
+        await this.ClearAssociatedItems(context);
+
+        this.SupplierItemRelations = associatedItems?.Select(item => new SupplierItemRelation()
+        {
+            ItemId = item.Id,
+            SupplierId = this.Id,
+        }).ToList();
+    }
+
+    public async Task<List<Item>> GetAssociatedItems(SharedContext context)
```
Comment (ChristofferGamel) [inline Models/Suppliers/Supplier.cs:48]: Comments :)

File: Controllers/Shared/ItemController.cs (hunk 3 above)
Comment (Thekid2002) [reply Controllers/Shared/ItemController.cs]: 🤮

**Review by ChristofferGamel — APPROVED** [review]
:+1:

**Pull request merged**